cmd/juju/charmhub/download.go

```diff
@@ -210,17 +222,29 @@ func (c *downloadCommand) Run(cmdContext *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
-	var (
-		found    bool
-		revision transport.InfoRevision
-	)
-	// If there is no series, then attempt to select the best one first.
-	if c.series == "" {
-		revision, found = c.locateRevisionByChannel(info.ChannelMap, charmChannel)
-	} else {
-		// We have a series go attempt to find the channel.
-		revision, found = c.locateRevisionByChannelAndSeries(info.ChannelMap, charmChannel, c.series)
+	var filterFn FilterInfoChannelMapFunc
+	switch {
+	case c.arch == ArchAll && c.series == SeriesAll:
+		filterFn = filterByArchitecture(c.fallbackArch)
```

Comment:
See my previous comment. I think that having the host as the fallback arch is wrong here. Could we add a check for the number of results and error with a message like "found multiple arch matches for X (arch1, arch2, arch3); please explicitly specify which arch to download via the --arch flag"